### PR TITLE
Perform restrained minimization on InitialStates

### PR DESCRIPTION
**Pull request by badisa**
* Address ligand poses moving too far from the original poses.

### CODE REVIEW

File: timemachine/potentials/bonded.py [hunk 1]
```diff
@@ -244,3 +245,17 @@ def log_flat_bottom_bond(conf, params, box, bond_idxs, beta):
...
     # so that energies have units of kJ/mol
     return jnp.sum(log_nrgs) / beta
+
+
+def harmonic_positional_restraint(x_init: Array, x_new: Array, box: Array, k: float = 4_000) -> Array:
+    r"""Harmonic positional restraint useful for performing minimization to prevent initial conformations
+    from changing too much.
+
+    This implements a harmonic bond potential, while being PBC aware:
+        V(x_init, x_init, k) = \sum k / 2 * sum((x_new - x_init)^2)
```
Comment (maxentile) [inline timemachine/potentials/bonded.py:255]: doc typo
```suggestion
        V(x_new, x_init, k) = \sum k / 2 * sum((x_new - x_init)^2)
```

File: timemachine/md/minimizer.py [hunk 2]
```diff
@@ -451,8 +453,35 @@ def val_and_grad_fn(coords):
...
+    x0: NDArray,
+    box0: NDArray,
+    free_idxs: NDArray,
+    k: float,
+) -> Callable[[NDArray], Tuple[float, NDArray]]:
+    restraint_val_and_grad = jax.value_and_grad(harmonic_positional_restraint, argnums=1)
+
+    starting_free = np.array(x0[free_idxs])
+
+    def wrapped_minimization(x):
```
Comment (maxentile) [inline timemachine/md/minimizer.py:467]: naming: `wrapped_val_and_grad` or similar

File: timemachine/potentials/bonded.py [hunk 3]
```diff
@@ -244,3 +245,17 @@ def log_flat_bottom_bond(conf, params, box, bond_idxs, beta):
     # note the extra 1/beta is to be consistent with other potentials
     # so that energies have units of kJ/mol
     return jnp.sum(log_nrgs) / beta
+
+
+def harmonic_positional_restraint(x_init: Array, x_new: Array, box: Array, k: float = 4_000) -> Array:
```
Comment (maxentile) [inline timemachine/potentials/bonded.py:250]: optional: `4_000` -> `DEFAULT_POSITIONAL_RESTRAINT_K`

**Review by maxentile — APPROVED** [review]

File: timemachine/potentials/bonded.py (hunk 1 above)
Comment (maxentile) [reply timemachine/potentials/bonded.py]: Oops, earlier suggestion might be [backward](https://github.com/proteneer/timemachine/pull/1393/files#diff-bb2d19446b7ea37b639b8b5a06a398d7be930ad81e47d67e143e2da183f07e1cR463):
```suggestion
        V(x_init, x_new, k) = \sum k / 2 * sum((x_new - x_init)^2)
```

File: timemachine/md/minimizer.py (hunk 2 above)
Comment (badisa) [reply timemachine/md/minimizer.py]: Changed in https://github.com/proteneer/timemachine/pull/1393/commits/8ec202fad2f1345b47d300d1a5f4104ea2e23673

File: timemachine/potentials/bonded.py (hunk 3 above)
Comment (badisa) [reply timemachine/potentials/bonded.py]: Done in https://github.com/proteneer/timemachine/pull/1393/commits/7388c815cf171e3cb01b3528215b7ed4b5a4c537

Also bumps the value back to 4000.0 to resolve an issue discussed offline.

**Pull request merged**